src/main.rs

```diff
@@ -287,6 +296,18 @@ fn settings() -> Result<Settings, Failure> {
         |path| Some(PathBuf::from(path)),
     );
 
+    // Read the config file path.
+    let output_path = matches.value_of(OUTPUT_DIR_OPTION).map_or_else(
+        || {
+            let mut candidate_dir = toastfile_path.to_path_buf();
+            candidate_dir.pop();
+            return Ok(candidate_dir);
+        },
+        |path| {
+            return Ok(PathBuf::from(path));
```

Comment:
If this code returns `Ok` in both branches of the `map_or_else`, it seems there is no point in wrapping it in a `Result`. Might as well just return the path and remove the `?` from line 309.

Separately: you need to explicitly use `return`. Just have the body of the closure evaluate to the path.